frontend/src/component/feature/FeatureView/FeatureOverview/ReleasePlan/ChangeRequest/RemoveReleasePlanChangeRequestDialog.tsx

```diff
@@ -0,0 +1,88 @@
+import { Dialogue } from 'component/common/Dialogue/Dialogue';
+import useToast from 'hooks/useToast';
+import { styled, Button, Alert } from '@mui/material';
+import type { IReleasePlan } from 'interfaces/releasePlans';
+import { useChangeRequestApi } from 'hooks/api/actions/useChangeRequestApi/useChangeRequestApi';
+import { usePendingChangeRequests } from 'hooks/api/getters/usePendingChangeRequests/usePendingChangeRequests';
+
+const StyledBoldSpan = styled('span')(({ theme }) => ({
+    fontWeight: theme.typography.fontWeightBold,
+}));
+
+interface IRemoveReleasePlanChangeRequestDialogProps {
+    projectId: string;
+    featureId: string;
+    environmentId: string;
+    releasePlan?: IReleasePlan | undefined;
+    environmentActive: boolean;
+    onClosing: () => void;
+}
+
+export const RemoveReleasePlanChangeRequestDialog = ({
+    projectId,
+    featureId,
+    environmentId,
+    releasePlan,
+    environmentActive,
+    onClosing,
+}: IRemoveReleasePlanChangeRequestDialogProps) => {
+    const { setToastData } = useToast();
+    const { addChange } = useChangeRequestApi();
+    const { refetch: refetchChangeRequests } =
+        usePendingChangeRequests(projectId);
+
+    const addRemoveReleasePlanToChangeRequest = async () => {
+        addChange(projectId, environmentId, {
+            feature: featureId,
+            action: 'deleteReleasePlan',
+            payload: {
+                planId: releasePlan?.id,
+            },
+        });
+
+        refetchChangeRequests();
+
+        setToastData({
+            type: 'success',
+            text: 'Added to draft',
+        });
+        onClosing();
+    };
```

Comment:
IMO the caller should own this. I think dialogs should only care about displaying the required information and emitting the "confirmation" callback to the parent, which then handles it.